### ISSUE TITLE
Add analysis-buddy

### ISSUE THREAD

**Issue by fengb** [fengb/zigbot9001#4]

**Comment by alexnask** [comment]
I just added the public API for use as a library.  
Small example:  
```zig
const analysis = @Import("analysis-buddy");

// ...
var prepared = try analysis.prepare(gpa, std_lib_path);
defer analysis.dispose(&prepared);

// Send a request
if (try analysis.analyse(&arena, &prepared, "std.io.Reader", std_lib_path)) |match| {
    // match is the URL
} else {
    // no match
}
```

**Comment by fengb** [comment]
Landed in 3112b6b0afc9f1dc754907cf502dd7ee243556b6
